File: src/main/java/ru/ioleksiv/telegram/bot/api/model/objects/std/keyboard/InlineKeyboardMarkup.java
package ru.ioleksiv.telegram.bot.api.model.objects.std.keyboard;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import ru.ioleksiv.telegram.bot.api.model.objects.std.keyboard.button.ButtonRow;
import ru.ioleksiv.telegram.bot.api.model.objects.std.keyboard.button.InlineKeyboardButton;

import java.util.Collection;
import java.util.List;

/**
 * @see <a href="https://core.telegram.org/bots/api#inlinekeyboardmarkup">InlineKeyboardMarkup</a>
 */
public class InlineKeyboardMarkup extends AbstractKeyboardMarkup<InlineKeyboardButton> {
    private static final Logger LOGGER = LoggerFactory.getLogger(InlineKeyboardMarkup.class);

    /**
     * inline_keyboard	Array of Array of InlineKeyboardButton	Array of button rows, each
     * represented by an Array of InlineKeyboardButton objects
     */

    @Override
    @JsonProperty("inline_keyboard")
    public List<ButtonRow<InlineKeyboardButton>> getRows() {
        return super.getRows();
    }

    @Override
    @JsonProperty("inline_keyboard")
    public void setRows(Collection<ButtonRow<InlineKeyboardButton>> rows) {
        super.setRows(rows);
    }

}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/objects/std/keyboard/InlineKeyboardMarkup.kt
package io.github.oleksivio.tl.kbot.server.api.objects.std.keyboard

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.objects.std.keyboard.row.InlineButtonRow

/**
 *  [InlineKeyboardMarkup](https://core.telegram.org/bots/api/#inlinekeyboardmarkup)
 */
data class InlineKeyboardMarkup(
        /**
         * inline_keyboard Array of Array of InlineKeyboardButton Array of button rows, each represented by an Array of
         * InlineKeyboardButton objects
         */
        @JsonProperty("inline_keyboard")
        var inlineKeyboard: ArrayList<InlineButtonRow> = ArrayList()

) : IKeyboard {
    fun add(row: InlineButtonRow) {
        inlineKeyboard.add(row)
    }
}
